src/server/models/cacheable_queries/opt-out.js

```diff
@@ -0,0 +1,98 @@
+import { r, OptOut } from '../../models'
+
+// STRUCTURE
+// maybe HASH by organization, so optout-<organization_id> has a <cell> key
+
+const orgCacheKey = (orgId) => `${process.env.CACHE_PREFIX|""}optouts-${orgId}`
+
+const loadMany = async (organizationId) => {
+  if (r.redis) {
+    const dbResult = await r.knex('opt_out')
+      .select('cell')
+      .where('organization_id', organizationId)
+    const cellOptOuts = dbResult.map((rec) => rec.cell)
+    const hashKey = orgCacheKey(organizationId)
+    // save 100 at a time
+    for (let i100=0, l100=Math.ceil(cellOptOuts.length/100); i100<l100; i100++) {
+      await r.redis.saddAsync(hashKey, cellOptOuts.slice(100*i100, 100*i100 + 100))
+    }
+    await r.redis.expire(hashKey, 86400)
+    console.log(`CACHE: Loaded optouts for ${organizationId}`)
+  }
+}
+
+export const optOutCache = {
+  clearQuery: async ({cell, organizationId}) => {
+    // remove cache by organization
+    // (if no cell is present, then clear whole query of organization)
+    if (r.redis) {
+      if (cell) {
+        await r.redis.sdelAsync(orgCacheKey(organizationId), cell)
```

Comment:
Based of the [documentation for redis sets](https://redis.io/commands#set), use `SREM` to delete a member of a set.